src/auth/oauth2/qgsauthoauth2method.cpp

```diff
@@ -142,23 +143,50 @@ bool QgsAuthOAuth2Method::updateNetworkRequest( QNetworkRequest &request, const
 
     if ( expired )
     {
-      msg = QStringLiteral( "Token expired, attempting refresh for authcfg %1" ).arg( authcfg );
-      QgsMessageLog::logMessage( msg, AUTH_METHOD_KEY, Qgis::MessageLevel::Info );
-
-      // Try to get a refresh token first
-      // go into local event loop and wait for a fired refresh-related slot
-      QEventLoop rloop( nullptr );
-      connect( o2, &QgsO2::refreshFinished, &rloop, &QEventLoop::quit );
-
-      // Asynchronously attempt the refresh
-      // TODO: This already has a timed reply setup in O2 base class (and in QgsNetworkAccessManager!)
-      //       May need to address this or app crashes will occur!
-      o2->refresh();
-
-      // block request update until asynchronous linking loop is quit
-      rloop.exec( QEventLoop::ExcludeUserInputEvents );
-
-      // refresh result should set o2 to (un)linked
+      if ( o2->refreshToken().isEmpty() || o2->refreshTokenUrl().isEmpty() )
```

Comment:
My understanding is that it's NOT a requirement to have an explicit refresh url for o2.

Reply:
The O2 library checks for these as well. I am doing an up-front check here, so the library does not unnecessarily throw a network error during the aborted refresh operation:
https://github.com/qgis/QGIS/blob/master/external/o2/src/o2.cpp#L412-L421

Reply:
Ah - i see what's happening. If an explicit refresh url isn't specified, o2 sets it to the token url. So all good here.